Packs/ApiModules/Scripts/AWSApiModule/AWSApiModule.py

```diff
@@ -0,0 +1,139 @@
+from CommonServerPython import *
+from CommonServerUserPython import *
+import boto3
+from botocore.config import Config
+
+
+def get_timeout(timeout):
+    if not timeout:
+        timeout = "60,10"  # default values
+    timeout_vals = timeout.split(',')
+    read_timeout = int(timeout_vals[0])
+    connect_timeout = 10 if len(timeout_vals) == 1 else int(timeout_vals[1])
+    return read_timeout, connect_timeout
+
+
+class AWSClient:
+
+    def __init__(self, aws_default_region, aws_role_arn, aws_role_session_name, aws_role_session_duration,
+                 aws_role_policy, aws_access_key_id, aws_secret_access_key, verify_certificate, timeout, retries):
+        self.aws_default_region = aws_default_region
+        self.aws_role_arn = aws_role_arn
+        self.aws_role_session_name = aws_role_session_name
+        self.aws_role_session_duration = aws_role_session_duration
+        self.aws_role_policy = aws_role_policy
+        self.aws_access_key_id = aws_access_key_id
+        self.aws_secret_access_key = aws_secret_access_key
+        self.verify_certificate = verify_certificate
+
+        proxies = handle_proxy(proxy_param_name='proxy', checkbox_default_value=False)
+        (read_timeout, connect_timeout) = get_timeout(timeout)
+        self.config = Config(
+            connect_timeout=connect_timeout,
+            read_timeout=read_timeout,
+            retries=dict(
+                max_attempts=int(retries)
+            ),
+            proxies=proxies
+        )
+
+    """HELPER FUNCTIONS"""
+
+    def aws_session(self, service, region=None, role_arn=None, role_session_name=None, role_session_duration=None,
```

Comment:
add type annotations.
also - all of theses args override the class props if given? let's add docstrings and explain that.